### ISSUE TITLE
coq_8_10 broken on darwin

### ISSUE THREAD

**Issue by Ptival** [NixOS/nixpkgs#66217]
**Describe the bug**
On nixpkgs-unstable (and I believe current master), `coq_8_10` fails to build on darwin:

```shell
[...]
OCAMLC    ide/coqide_os_specific.mli
OCAMLOPT  ide/coqide_os_specific.ml
OCAMLOPT -o bin/coqide
ld: framework not found Cocoa
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
File "caml_startup", line 1:
Error: Error during linking
```

**Metadata**
```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.6.0, macOS 10.14.5`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `""`
 - channels(val): `"nixpkgs-19.09pre187222.002b853782e"`
 - nixpkgs: `/Users/val/.nix-defexpr/channels/nixpkgs`
```

**Comment by Ptival** [comment]
Conditionally adding `darwin.apple_sdk.frameworks.Cocoa` makes it give another error message, this time a linker error seemingly about GTK3 symbols.

I am now trying to build it with `gtk3` added as a build input, but I wonder if CoqIDE even builds on non-darwin architectures...

**Comment by Ptival** [comment]
Not sure how to solve the next problem, adding `gtk3` is not enough, I suspect there has to be some linker flag setup that I am not aware of. Current error is:
```shell
OCAMLOPT -o bin/coqide
Undefined symbols for architecture x86_64:
  "_GTK_SOCKET", referenced from:
      _ml_gtk_socket_add_id in liblablgtk3_stubs.a(ml_gtk.o)
      _ml_gtk_socket_get_id in liblablgtk3_stubs.a(ml_gtk.o)
      _ml_gtk_socket_get_plug_window in liblablgtk3_stubs.a(ml_gtk.o)
  "_gtk_plug_get_type", referenced from:
      _ml_gtkwindow_init in liblablgtk3_stubs.a(ml_gtk.o)
  "_gtk_plug_new", referenced from:
      _ml_gtk_plug_new in liblablgtk3_stubs.a(ml_gtk.o)
     (maybe you meant: _ml_gtk_plug_new)
  "_gtk_socket_add_id", referenced from:
      _ml_gtk_socket_add_id in liblablgtk3_stubs.a(ml_gtk.o)
     (maybe you meant: _ml_gtk_socket_add_id)
  "_gtk_socket_get_id", referenced from:
      _ml_gtk_socket_get_id in liblablgtk3_stubs.a(ml_gtk.o)
     (maybe you meant: _ml_gtk_socket_get_id)
  "_gtk_socket_get_plug_window", referenced from:
      _ml_gtk_socket_get_plug_window in liblablgtk3_stubs.a(ml_gtk.o)
     (maybe you meant: _ml_gtk_socket_get_plug_window)
  "_gtk_socket_get_type", referenced from:
      _ml_gtkwindow_init in liblablgtk3_stubs.a(ml_gtk.o)
ld: symbol(s) not found for architecture x86_64
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
File "caml_startup", line 1:
Error: Error during linking
make[1]: *** [Makefile.ide:108: bin/coqide] Error 2
```

**Comment by vbgl** [comment]
This is a bug in the build system of gtk+-3.0 at 3.24.10. It has been fixed in 3.24.11. See discussion there: https://github.com/garrigue/lablgtk/issues/82

However, there is an other bug of the build system at 3.24.11. This one affects all platforms: https://gitlab.gnome.org/GNOME/gtk/issues/2136
